In [12]:
import pandas as pd

df= pd.read_excel(r'C:\Users\Amr\Downloads\Customer Call List.xlsx')
df

CustomerID First_Name    Last_Name  Phone_Number  \
0         1001      Frodo      Baggins  123-545-5421   
1         1002       Abed        Nadir  123/643/9775   
2         1003     Walter       /White    7066950392   
3         1004     Dwight      Schrute  123-543-2345   
4         1005        Jon         Snow  876|678|3469   
5         1006        Ron      Swanson  304-762-2467   
6         1007       Jeff       Winger           NaN   
7         1008   Sherlock       Holmes  876|678|3469   
8         1009    Gandalf          NaN           N/a   
9         1010      Peter       Parker  123-545-5421   
10        1011    Samwise       Gamgee           NaN   
11        1012      Harry    ...Potter    7066950392   
12        1013        Don       Draper  123-543-2345   
13        1014     Leslie        Knope  876|678|3469   
14        1015       Toby  Flenderson_  304-762-2467   
15        1016        Ron      Weasley  123-545-5421   
16        1017   Michael         Scott  123/643/9775   
17        1018      Clark         Kent    7066950392   
18        1019      Creed       Braton           N/a   
19        1020     Anakin    Skywalker  876|678|3469   
20        1020     Anakin    Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   
20            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True  
20               True

### Dropping Duplicates

In [13]:
df.drop_duplicates(inplace=True)
df

CustomerID First_Name    Last_Name  Phone_Number  \
0         1001      Frodo      Baggins  123-545-5421   
1         1002       Abed        Nadir  123/643/9775   
2         1003     Walter       /White    7066950392   
3         1004     Dwight      Schrute  123-543-2345   
4         1005        Jon         Snow  876|678|3469   
5         1006        Ron      Swanson  304-762-2467   
6         1007       Jeff       Winger           NaN   
7         1008   Sherlock       Holmes  876|678|3469   
8         1009    Gandalf          NaN           N/a   
9         1010      Peter       Parker  123-545-5421   
10        1011    Samwise       Gamgee           NaN   
11        1012      Harry    ...Potter    7066950392   
12        1013        Don       Draper  123-543-2345   
13        1014     Leslie        Knope  876|678|3469   
14        1015       Toby  Flenderson_  304-762-2467   
15        1016        Ron      Weasley  123-545-5421   
16        1017   Michael         Scott  123/643/9775   
17        1018      Clark         Kent    7066950392   
18        1019      Creed       Braton           N/a   
19        1020     Anakin    Skywalker  876|678|3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True

### Cleaning `Last_Name`

If you have a column in a pandas DataFrame that contains unwanted letters and you want to remove them, you can use the `.str.replace(r'[]', '')` function with regex.

In [14]:
df['Last_Name'].str.replace(r'[/, ... ,_]', '', regex=True) # regex regular expression

0        Baggins
1          Nadir
2          White
3        Schrute
4           Snow
5        Swanson
6         Winger
7         Holmes
8            NaN
9         Parker
10        Gamgee
11        Potter
12        Draper
13         Knope
14    Flenderson
15       Weasley
16         Scott
17          Kent
18        Braton
19     Skywalker
Name: Last_Name, dtype: object

Or `.str.strip('')` can used

In [15]:
df['Last_Name'].str.strip ('123./_') 


0        Baggins
1          Nadir
2          White
3        Schrute
4           Snow
5        Swanson
6         Winger
7         Holmes
8            NaN
9         Parker
10        Gamgee
11        Potter
12        Draper
13         Knope
14    Flenderson
15       Weasley
16         Scott
17          Kent
18        Braton
19     Skywalker
Name: Last_Name, dtype: object

In [16]:
#df['Last_Name'].str.strip ('/') # strip removes from the edges not the middle
# df['Last_Name'].str.strip ('...')
# df['Last_Name'].str.strip ('_')
df['Last_Name'] = df['Last_Name'].str.strip ('123./_')    # removing what's inside the ''

### Cleaning `Phone_Number`

In [77]:
df['Phone_Number'].str.replace('[^0-9]','', regex=True) # this would get me NaN in index 2 beacause it's int not a str

0     1235455421
1     1236439775
2     7066950392
3     1235432345
4     8766783469
5     3047622467
6               
7     8766783469
8               
9     1235455421
10              
11    7066950392
12    1235432345
13    8766783469
14    3047622467
15    1235455421
16    1236439775
17    7066950392
18              
19    8766783469
Name: Phone_Number, dtype: object

In [68]:
type(df.loc[2,'Phone_Number'])

int

- `.astype(str)` changes the series to a str type for all values like the value `7066950392` from `int` to `str`
- `.str.replace('[^a-zA-Z0-9]','', regex=True)` replaces anything other than the capital and small letters and also the numbers via `^`

In [17]:
df['Phone_Number'] = df['Phone_Number'].astype(str).str.replace('[^a-zA-Z0-9]','', regex=True)   # astype(str) turn the values in phone_Number into strings to be able to use str.replace on them

In [18]:
df

CustomerID First_Name   Last_Name Phone_Number  \
0         1001      Frodo     Baggins   1235455421   
1         1002       Abed       Nadir   1236439775   
2         1003     Walter       White   7066950392   
3         1004     Dwight     Schrute   1235432345   
4         1005        Jon        Snow   8766783469   
5         1006        Ron     Swanson   3047622467   
6         1007       Jeff      Winger          nan   
7         1008   Sherlock      Holmes   8766783469   
8         1009    Gandalf         NaN           Na   
9         1010      Peter      Parker   1235455421   
10        1011    Samwise      Gamgee          nan   
11        1012      Harry      Potter   7066950392   
12        1013        Don      Draper   1235432345   
13        1014     Leslie       Knope   8766783469   
14        1015       Toby  Flenderson   3047622467   
15        1016        Ron     Weasley   1235455421   
16        1017   Michael        Scott   1236439775   
17        1018      Clark        Kent   7066950392   
18        1019      Creed      Braton           Na   
19        1020     Anakin   Skywalker   8766783469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True

In [8]:
df['Phone_Number'].apply(lambda x: x[0:3] +'-'+ x[3:6] +'-'+ x[6:] if x not in ['nan', 'Na'] else x) # because the column values is a str I can use slicing 
# if I hadn't used astype(str) before I would've done it in this code with str(x)

0     123-545-5421
1     123-643-9775
2     706-695-0392
3     123-543-2345
4     876-678-3469
5     304-762-2467
6              nan
7     876-678-3469
8               Na
9     123-545-5421
10             nan
11    706-695-0392
12    123-543-2345
13    876-678-3469
14    304-762-2467
15    123-545-5421
16    123-643-9775
17    706-695-0392
18              Na
19    876-678-3469
Name: Phone_Number, dtype: object

In [ ]:
df['Phone_Number'] = df['Phone_Number'].apply(lambda x: x[0:3] +'-'+ x[3:6] +'-'+ x[6:] if x not in ['nan', 'Na'] else x)

df['Phone_Number'] = df['Phone_Number'].str.replace(r'[nan, Na]' , '', regex=True)


In [20]:
df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed       Nadir  123-643-9775   
2         1003     Walter       White  706-695-0392   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876-678-3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger                 
7         1008   Sherlock      Holmes  876-678-3469   
8         1009    Gandalf         NaN                 
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee                 
11        1012      Harry      Potter  706-695-0392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876-678-3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123-643-9775   
17        1018      Clark        Kent  706-695-0392   
18        1019      Creed      Braton                 
19        1020     Anakin   Skywalker  876-678-3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire             Yes             No   
1                     93 West Main Street              No            Yes   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503             Yes              Y   
4                        123 Dragons Road               Y             No   
5                        768 City Parkway             Yes            Yes   
6                       1209 South Street              No             No   
7                           98 Clue Drive               N             No   
8                        123 Middle Earth             Yes            NaN   
9              25th Main Street, New York             Yes             No   
10                  612 Shire Lane, Shire             Yes             No   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street             Yes              N   
13                       343 City Parkway             Yes             No   
14                          214 HR Avenue               N             No   
15                   2395 Hogwarts Avenue              No              N   
16         121 Paper Avenue, Pennsylvania             Yes             No   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a            Yes   
19            910 Tatooine Road, Tatooine             Yes              N   

    Not_Useful_Column  
0                True  
1               False  
2                True  
3                True  
4                True  
5                True  
6               False  
7               False  
8               False  
9                True  
10               True  
11               True  
12              False  
13              False  
14              False  
15              False  
16              False  
17               True  
18               True  
19               True

### Cleaning `Address`

`str.split(sep= , expand=)`

In [27]:
df[['Street_Address', 'State', 'Zip']] = df['Address'].str.split(',', expand=True) # spliting based on seperator

In [ ]:
df['Paying Customer'] = df['Paying Customer'].str.replace('Yes', 'Y')

df['Paying Customer'] = df['Paying Customer'].str.replace('No', 'N')


In [ ]:
df['Do_Not_Contact'] = df['Do_Not_Contact'].str.replace('Yes', 'Y')

df['Do_Not_Contact'] = df['Do_Not_Contact'].str.replace('No', 'N')


In [37]:
df.drop(columns= 'Not_Useful_Column', inplace=True)

In [38]:
df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed       Nadir  123-643-9775   
2         1003     Walter       White  706-695-0392   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876-678-3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger                 
7         1008   Sherlock      Holmes  876-678-3469   
8         1009    Gandalf         NaN                 
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee                 
11        1012      Harry      Potter  706-695-0392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876-678-3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123-643-9775   
17        1018      Clark        Kent  706-695-0392   
18        1019      Creed      Braton                 
19        1020     Anakin   Skywalker  876-678-3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire               Y              N   
1                     93 West Main Street               N              Y   
2                      298 Drugs Driveway               N            NaN   
3   980 Paper Avenue, Pennsylvania, 18503               Y              Y   
4                        123 Dragons Road               Y              N   
5                        768 City Parkway               Y              Y   
6                       1209 South Street               N              N   
7                           98 Clue Drive               N              N   
8                        123 Middle Earth               Y            NaN   
9              25th Main Street, New York               Y              N   
10                  612 Shire Lane, Shire               Y              N   
11                   2394 Hogwarts Avenue               Y            NaN   
12                       2039 Main Street               Y              N   
13                       343 City Parkway               Y              N   
14                          214 HR Avenue               N              N   
15                   2395 Hogwarts Avenue               N              N   
16         121 Paper Avenue, Pennsylvania               Y              N   
17                        3498 Super Lane               Y            NaN   
18                                    N/a             N/a              Y   
19            910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State     Zip  
0         123 Shire Lane          Shire    None  
1    93 West Main Street           None    None  
2     298 Drugs Driveway           None    None  
3       980 Paper Avenue   Pennsylvania   18503  
4       123 Dragons Road           None    None  
5       768 City Parkway           None    None  
6      1209 South Street           None    None  
7          98 Clue Drive           None    None  
8       123 Middle Earth           None    None  
9       25th Main Street       New York    None  
10        612 Shire Lane          Shire    None  
11  2394 Hogwarts Avenue           None    None  
12      2039 Main Street           None    None  
13      343 City Parkway           None    None  
14         214 HR Avenue           None    None  
15  2395 Hogwarts Avenue           None    None  
16      121 Paper Avenue   Pennsylvania    None  
17       3498 Super Lane           None    None  
18                   N/a           None    None  
19     910 Tatooine Road       Tatooine    None

### Dropping NaN

Don't use `.replace('[NaN,Na,N/a]','')` as Square brackets [] define a character class, meaning it looks for any of the characters `N`, `a`, `/`, or `,` individually.
- Use `.replace['','']`

`replace()` does not handle `NaN` unless it's explicitly converted to a string.
If the DataFrame contains actual `NaN` values (not strings), use:
- `df.replace([np.nan, 'Na', 'N/a'], '')` or remove NaN by `fillna('')` then user the first replace and it will work.

In [ ]:
import numpy as np

df = df.replace([np.NaN,'Na','N/a'],'', regex=True)
df

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1002       Abed         dir  123-643-9775   
2         1003     Walter       White  706-695-0392   
3         1004     Dwight     Schrute  123-543-2345   
4         1005        Jon        Snow  876-678-3469   
5         1006        Ron     Swanson  304-762-2467   
6         1007       Jeff      Winger                 
7         1008   Sherlock      Holmes  876-678-3469   
8         1009    Gandalf                             
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee                 
11        1012      Harry      Potter  706-695-0392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876-678-3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123-643-9775   
17        1018      Clark        Kent  706-695-0392   
18        1019      Creed      Braton                 
19        1020     Anakin   Skywalker  876-678-3469   

                                  Address Paying Customer Do_Not_Contact  \
0                   123 Shire Lane, Shire               Y              N   
1                     93 West Main Street               N              Y   
2                      298 Drugs Driveway               N                  
3   980 Paper Avenue, Pennsylvania, 18503               Y              Y   
4                        123 Dragons Road               Y              N   
5                        768 City Parkway               Y              Y   
6                       1209 South Street               N              N   
7                           98 Clue Drive               N              N   
8                        123 Middle Earth               Y                  
9              25th Main Street, New York               Y              N   
10                  612 Shire Lane, Shire               Y              N   
11                   2394 Hogwarts Avenue               Y                  
12                       2039 Main Street               Y              N   
13                       343 City Parkway               Y              N   
14                          214 HR Avenue               N              N   
15                   2395 Hogwarts Avenue               N              N   
16         121 Paper Avenue, Pennsylvania               Y              N   
17                        3498 Super Lane               Y                  
18                                                                     Y   
19            910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State     Zip  
0         123 Shire Lane          Shire          
1    93 West Main Street                         
2     298 Drugs Driveway                         
3       980 Paper Avenue   Pennsylvania   18503  
4       123 Dragons Road                         
5       768 City Parkway                         
6      1209 South Street                         
7          98 Clue Drive                         
8       123 Middle Earth                         
9       25th Main Street       New York          
10        612 Shire Lane          Shire          
11  2394 Hogwarts Avenue                         
12      2039 Main Street                         
13      343 City Parkway                         
14         214 HR Avenue                         
15  2395 Hogwarts Avenue                         
16      121 Paper Avenue   Pennsylvania          
17       3498 Super Lane                         
18                                               
19     910 Tatooine Road       Tatooine

### Dropping rows that has the value `Do_Not_Contact` is a `Y`


If the column has more than 2 values then this code is better but in my case the first one is feasible

In [ ]:

for x in df.index:
    if df.loc[x, 'Do_Not_Contact'] == 'Y':
        df.drop(x, inplace=True)


In [ ]:
# or 
df_Customer_Call_List = df[df['Do_Not_Contact'] != 'Y']
df_Customer_Call_List

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
2         1003     Walter       White  706-695-0392   
4         1005        Jon        Snow  876-678-3469   
6         1007       Jeff      Winger                 
7         1008   Sherlock      Holmes  876-678-3469   
8         1009    Gandalf                             
9         1010      Peter      Parker  123-545-5421   
10        1011    Samwise      Gamgee                 
11        1012      Harry      Potter  706-695-0392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876-678-3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123-643-9775   
17        1018      Clark        Kent  706-695-0392   
19        1020     Anakin   Skywalker  876-678-3469   

                           Address Paying Customer Do_Not_Contact  \
0            123 Shire Lane, Shire               Y              N   
2               298 Drugs Driveway               N                  
4                 123 Dragons Road               Y              N   
6                1209 South Street               N              N   
7                    98 Clue Drive               N              N   
8                 123 Middle Earth               Y                  
9       25th Main Street, New York               Y              N   
10           612 Shire Lane, Shire               Y              N   
11            2394 Hogwarts Avenue               Y                  
12                2039 Main Street               Y              N   
13                343 City Parkway               Y              N   
14                   214 HR Avenue               N              N   
15            2395 Hogwarts Avenue               N              N   
16  121 Paper Avenue, Pennsylvania               Y              N   
17                 3498 Super Lane               Y                  
19     910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State Zip  
0         123 Shire Lane          Shire      
2     298 Drugs Driveway                     
4       123 Dragons Road                     
6      1209 South Street                     
7          98 Clue Drive                     
8       123 Middle Earth                     
9       25th Main Street       New York      
10        612 Shire Lane          Shire      
11  2394 Hogwarts Avenue                     
12      2039 Main Street                     
13      343 City Parkway                     
14         214 HR Avenue                     
15  2395 Hogwarts Avenue                     
16      121 Paper Avenue   Pennsylvania      
17       3498 Super Lane                     
19     910 Tatooine Road       Tatooine

In [63]:
df_Customer_Call_List = df_Customer_Call_List[df_Customer_Call_List['Phone_Number'] != '']

# or
# for x in df.index:
#     if df.loc[x, 'Do_Not_Contact'] == '':
#         df.drop(x, inplace=True)

df_Customer_Call_List 

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
2         1003     Walter       White  706-695-0392   
4         1005        Jon        Snow  876-678-3469   
7         1008   Sherlock      Holmes  876-678-3469   
9         1010      Peter      Parker  123-545-5421   
11        1012      Harry      Potter  706-695-0392   
12        1013        Don      Draper  123-543-2345   
13        1014     Leslie       Knope  876-678-3469   
14        1015       Toby  Flenderson  304-762-2467   
15        1016        Ron     Weasley  123-545-5421   
16        1017   Michael        Scott  123-643-9775   
17        1018      Clark        Kent  706-695-0392   
19        1020     Anakin   Skywalker  876-678-3469   

                           Address Paying Customer Do_Not_Contact  \
0            123 Shire Lane, Shire               Y              N   
2               298 Drugs Driveway               N                  
4                 123 Dragons Road               Y              N   
7                    98 Clue Drive               N              N   
9       25th Main Street, New York               Y              N   
11            2394 Hogwarts Avenue               Y                  
12                2039 Main Street               Y              N   
13                343 City Parkway               Y              N   
14                   214 HR Avenue               N              N   
15            2395 Hogwarts Avenue               N              N   
16  121 Paper Avenue, Pennsylvania               Y              N   
17                 3498 Super Lane               Y                  
19     910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State Zip  
0         123 Shire Lane          Shire      
2     298 Drugs Driveway                     
4       123 Dragons Road                     
7          98 Clue Drive                     
9       25th Main Street       New York      
11  2394 Hogwarts Avenue                     
12      2039 Main Street                     
13      343 City Parkway                     
14         214 HR Avenue                     
15  2395 Hogwarts Avenue                     
16      121 Paper Avenue   Pennsylvania      
17       3498 Super Lane                     
19     910 Tatooine Road       Tatooine

In [80]:
df_Customer_Call_List['Do_Not_Contact'].str.replace([np.nan], 'N', regex=True)

TypeError: unhashable type: 'list'

In [ ]:
df_Customer_Call_List['Do_Not_Contact'].replace('', 'N')


0     N
2     N
4     N
7     N
9     N
11    N
12    N
13    N
14    N
15    N
16    N
17    N
19    N
Name: Do_Not_Contact, dtype: object

In [ ]:
df_Customer_Call_List['Do_Not_Contact'] = df_Customer_Call_List['Do_Not_Contact'].replace('', 'N')

In [ ]:
df_Customer_Call_List.reset_index(drop=True, inplace=True)


CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1003     Walter       White  706-695-0392   
2         1005        Jon        Snow  876-678-3469   
3         1008   Sherlock      Holmes  876-678-3469   
4         1010      Peter      Parker  123-545-5421   
5         1012      Harry      Potter  706-695-0392   
6         1013        Don      Draper  123-543-2345   
7         1014     Leslie       Knope  876-678-3469   
8         1015       Toby  Flenderson  304-762-2467   
9         1016        Ron     Weasley  123-545-5421   
10        1017   Michael        Scott  123-643-9775   
11        1018      Clark        Kent  706-695-0392   
12        1020     Anakin   Skywalker  876-678-3469   

                           Address Paying Customer Do_Not_Contact  \
0            123 Shire Lane, Shire               Y              N   
1               298 Drugs Driveway               N              N   
2                 123 Dragons Road               Y              N   
3                    98 Clue Drive               N              N   
4       25th Main Street, New York               Y              N   
5             2394 Hogwarts Avenue               Y              N   
6                 2039 Main Street               Y              N   
7                 343 City Parkway               Y              N   
8                    214 HR Avenue               N              N   
9             2395 Hogwarts Avenue               N              N   
10  121 Paper Avenue, Pennsylvania               Y              N   
11                 3498 Super Lane               Y              N   
12     910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State Zip  
0         123 Shire Lane          Shire      
1     298 Drugs Driveway                     
2       123 Dragons Road                     
3          98 Clue Drive                     
4       25th Main Street       New York      
5   2394 Hogwarts Avenue                     
6       2039 Main Street                     
7       343 City Parkway                     
8          214 HR Avenue                     
9   2395 Hogwarts Avenue                     
10      121 Paper Avenue   Pennsylvania      
11       3498 Super Lane                     
12     910 Tatooine Road       Tatooine

In [85]:
df_Customer_Call_List.drop(columns= 'Zip')

CustomerID First_Name   Last_Name  Phone_Number  \
0         1001      Frodo     Baggins  123-545-5421   
1         1003     Walter       White  706-695-0392   
2         1005        Jon        Snow  876-678-3469   
3         1008   Sherlock      Holmes  876-678-3469   
4         1010      Peter      Parker  123-545-5421   
5         1012      Harry      Potter  706-695-0392   
6         1013        Don      Draper  123-543-2345   
7         1014     Leslie       Knope  876-678-3469   
8         1015       Toby  Flenderson  304-762-2467   
9         1016        Ron     Weasley  123-545-5421   
10        1017   Michael        Scott  123-643-9775   
11        1018      Clark        Kent  706-695-0392   
12        1020     Anakin   Skywalker  876-678-3469   

                           Address Paying Customer Do_Not_Contact  \
0            123 Shire Lane, Shire               Y              N   
1               298 Drugs Driveway               N              N   
2                 123 Dragons Road               Y              N   
3                    98 Clue Drive               N              N   
4       25th Main Street, New York               Y              N   
5             2394 Hogwarts Avenue               Y              N   
6                 2039 Main Street               Y              N   
7                 343 City Parkway               Y              N   
8                    214 HR Avenue               N              N   
9             2395 Hogwarts Avenue               N              N   
10  121 Paper Avenue, Pennsylvania               Y              N   
11                 3498 Super Lane               Y              N   
12     910 Tatooine Road, Tatooine               Y              N   

          Street_Address          State  
0         123 Shire Lane          Shire  
1     298 Drugs Driveway                 
2       123 Dragons Road                 
3          98 Clue Drive                 
4       25th Main Street       New York  
5   2394 Hogwarts Avenue                 
6       2039 Main Street                 
7       343 City Parkway                 
8          214 HR Avenue                 
9   2395 Hogwarts Avenue                 
10      121 Paper Avenue   Pennsylvania  
11       3498 Super Lane                 
12     910 Tatooine Road       Tatooine